libmamba/include/mamba/specs/version.hpp

```diff
@@ -62,13 +62,44 @@ namespace mamba::specs
     /**
      * A sequence of VersionPartAtom meant to represent a part of a version (e.g. major, minor).
      *
+     * In a version like ``1.3.0post1dev``, the parts are ``1``, ``3``, and ``0post1dev``.
      * Version parts can have a arbitrary number of atoms, such as {0, "post"} {1, "dev"}
-     * in 0post1dev
+     * in ``0post1dev``.
      *
      * @see  Version::parse for how this is computed from strings.
      * @todo Use a small vector of expected size 1 if performance ar not good enough.
      */
-    using VersionPart = std::vector<VersionPartAtom>;
+    struct VersionPart
+    {
+        /** The atoms of the version part */
+        std::vector<VersionPartAtom> atoms = {};
+
+        /**
+         * Whether a potential leading zero in the first atom should be considered implicit.
+         *
+         * During parsing of ``Version``, if a part starts with a literal atom, it is considered
+         * the same as if it started with a leading ``0``.
+         * For instance ``0post1dev`` is parsed in the same way as ``post1dev``.
+         * Marking it as implicit enables the possibility to remove it when reconstructing a string
+         * representation.
+         * This is desirable for compatibility with other version formats, such as Python, where
+         * a version modifier might be expressed as ``1.3.0.dev3``.
+         */
+        bool implicit_leading_zero = false;
+
+        VersionPart();
+        VersionPart(std::initializer_list<VersionPartAtom> init);
+        VersionPart(std::vector<VersionPartAtom> atoms, bool implicit_leading_zero);
+
+        [[nodiscard]] auto str() const -> std::string;
+
+        auto operator==(const VersionPart& other) const -> bool;
```

Comment:
If it does make sense (not sure given the implementation), add a todo comment about replacing these operators by defaulted ones in C++20

Reply:
Absolutely not default ;)